Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -1734,6 +1727,13 @@ FloatSize MediaPlayerPrivateGStreamer::naturalSize() const
     if (!m_videoSize.isEmpty())
         return m_videoSize;
 
+#if USE(GSTREAMER_HOLEPUNCH)
```

Comment:
So, the main change (as I see it) about moving this block down is to give the opportunity to return a size when using MediaStream and an empty size when there's no video (the condition above), right?

Reply:
MSE player parses video size from gst caps and sets m_videoSize. so, this change provides the correct video size for MSE playbacks. for other players it should work as before.

Reply:
That was confusing me, as I wasn't able to see where m_videoSize would be set in the regular player private (the GSTREAMER_HOLEPUNCH code would prevent the "m_videoSize = ..." code down in the method to be executed. Now that I see that it can also be set from the MSE player private, your change completely makes sense.